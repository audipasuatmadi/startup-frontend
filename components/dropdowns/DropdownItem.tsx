import React from 'react';
import TextButton from '../buttons/TextButton';

export interface DropdownItemProps {
  name?: string;
  type?: 'button' | 'submit' | 'reset';
  className?: string;
  children?: string;
  disabled?: boolean;

  onClick?: (
    e: React.MouseEvent<HTMLButtonElement | HTMLAnchorElement>
  ) => void;
  href?: string;

  divider?: boolean
}

const DropdownItem = ({className, divider, ...otherProps}: DropdownItemProps) => {
  return (
    <li
      className={`list-none${
        divider? ' border-t-2 border-gray-300' : ''
      }`}>
      <TextButton
        className='block w-full'
        {...otherProps}
      />
    </li>
  );
};

export default DropdownItem;
