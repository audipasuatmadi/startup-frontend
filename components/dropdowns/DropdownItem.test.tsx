import React from 'react'
import { shallow } from 'enzyme'
import DropdownItem from './DropdownItem'

describe('DropdownItem rendering tests', () => {
  it('should render properly', () => {
    const wrapper = shallow(<DropdownItem />)
    expect(wrapper).toMatchSnapshot()
  })
})