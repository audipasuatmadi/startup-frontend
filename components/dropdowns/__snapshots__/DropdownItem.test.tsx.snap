// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownItem rendering tests should render properly 1`] = `
<li
  className="list-none"
>
  <TextButton
    className="block w-full"
    theme="brand"
  />
</li>
`;
